### PR TITLE
docs: durable subscription - downgrade history retention for indexes

### PR DESCRIPTION
**Pull request by kay-kim**
- Made the index a footnote.  Alternatively, can instead raise its prominence as a separate callout stating that it isn't recommended. 
  - If, in practice, people are doing it a lot, raising the prominence in a callout will be more effective.  
  - Else if, in practice, people haven't been doing it one way or the other, then, keeping a low profile will probably be more effective.

### CODE REVIEW

File: doc/user/content/transform-data/patterns/durable-subscriptions.md [hunk 1]
```diff
@@ -35,12 +34,12 @@ application following a connection disruption, you can:
 By default, all user-defined sources, tables, materialized views, and indexes
 keep track of the most recent version of their underlying data. To gracefully
 recover from connection disruptions and enable lossless, _durable
-subscriptions_, you can configure the objects the subscription depends on
-to **retain history**.
+subscriptions_, you can configure the sources, tables, and materialized views
+that the subscription depends on to **retain history**.[^1]
```
Comment (kay-kim) [inline doc/user/content/transform-data/patterns/durable-subscriptions.md:38]: https://preview.materialize.com/materialize/30230/transform-data/patterns/durable-subscriptions/#history-retention-period

File: doc/user/content/transform-data/patterns/durable-subscriptions.md [hunk 2]
```diff
@@ -82,9 +76,8 @@ increased resource utilization.  See [Considerations](#considerations).
 {{</ note >}}
 
 To set the history retention period for [sources](/sql/create-source/),
-[tables](/sql/create-table/), [materialized
-views](/sql/create-materialized-view/), and [indexes](/sql/create-index/), you
-can either:
+[tables](/sql/create-table/), and [materialized
+views](/sql/create-materialized-view/)[^1], you can either:
```
Comment (kay-kim) [inline doc/user/content/transform-data/patterns/durable-subscriptions.md:80]: https://preview.materialize.com/materialize/30230/transform-data/patterns/durable-subscriptions/#considerations

File: doc/user/content/transform-data/patterns/durable-subscriptions.md [hunk 3]
```diff
@@ -253,5 +233,6 @@ As a result, to guarantee that the data processing occurs only once after your
 application crashes, you must write the progress message `mz_timestamp` and all
 buffered data **together in a single transaction**.
 
-<!--TODO(chaas): add top-level section on point-in-time queries, similar to Durable
-subscriptions section-->
+[^1]: Configuring indexes to retain history is not fully supported and is not
```
Comment (kay-kim) [inline doc/user/content/transform-data/patterns/durable-subscriptions.md:236]: https://preview.materialize.com/materialize/30230/transform-data/patterns/durable-subscriptions/#fn:1

File: doc/user/content/transform-data/patterns/durable-subscriptions.md [hunk 4]
```diff
@@ -157,30 +150,18 @@ ALTER MATERIALIZED VIEW winning_bids RESET (RETAIN HISTORY);
...
-  index, consider creating a materialized view for your subscription query and
-  configuring the history retention period on that materialized view instead.
-
-- Similarly, because of the increased storage costs and processing time for the
-  additional historical data on whichever object has an increased history
-  retention period, consider configuring history retention period on the index
-  or materialized view directly powering the subscription, rather than all the
-  way through the dependency chain from the source to the index or materialized
-  view.
+Because of the increased storage costs and processing time for the additional
```
Comment (kay-kim) [inline doc/user/content/transform-data/patterns/durable-subscriptions.md:161]: https://preview.materialize.com/materialize/30230/transform-data/patterns/durable-subscriptions/#best-practices

File: doc/user/content/sql/subscribe.md [hunk 5]
```diff
@@ -132,10 +132,10 @@ earlier than the earliest historical state retained by the underlying objects,
 an error is thrown.
 
 To configure the history retention period for objects used in a subscription,
-see
-[Durable subscription](/transform-data/patterns/durable-subscriptions/#histo
-ry-retention-period). If `AS OF` is unspecified, the system automatically
-chooses an `AS OF` timestamp.
+see [Durable
+subscriptions](/transform-data/patterns/durable-subscriptions/#history-retention-period).
+If `AS OF` is unspecified, the system automatically chooses an `AS OF`
```
Comment (kay-kim) [inline doc/user/content/sql/subscribe.md:137]: https://preview.materialize.com/materialize/30230/sql/subscribe/#as-of

**Comment by frankmcsherry** [comment]
This checks out to me. The only nit I have is that saying "indexes are not fully supported" seems odd when the whole thing is in private preview and so also "not fully supported". I'm not sure there's a risk that folks will think that the other options are fully supported, but .. idk if there's a best practices around that.

**Review by frankmcsherry — APPROVED** [review]
I left a comment, but it is a nit more than an actionable input.

**Comment by kay-kim** [comment]
Will remove the fully supported part

**Review by chaas — APPROVED** [review]
Thanks, Kay!

**Pull request merged**